### PR TITLE
Fixes flaky inputs changes detection in `dy-static-file-server` services

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

Fix buggy inputs detection. Previous version would give up after a bit causing the CI to be flaky.

## Checklist

- [ ] I think the code is well written
- [x] I updated the table of contents (make toc)
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] **Runs in the swarm**
- [ ] If you design a new module, add your user to .github/CODEOWNERS

### CODE REVIEW

File: services/dy-static-file-server/docker/custom/Dockerfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ FROM joseluisq/static-web-server:2.0.2-alpine as production
 #     docker run dy-static-file-server:prod
 #
 
-ARG PYTHON_VERSION="3.8.10-r0"
+ARG PYTHON_VERSION="3.8.15-r0"
```
Comment (sanderegg) [inline services/dy-static-file-server/docker/custom/Dockerfile:10]: you do not want to go straight for 3.11?
also the version of your web-server is trailing way behind. we use 2.32 in simcore.

**Review by sanderegg — APPROVED** [review]

File: services/dy-static-file-server/docker/custom/Dockerfile (hunk 1 above)
Comment (GitHK) [reply services/dy-static-file-server/docker/custom/Dockerfile:10]: it depends on the version inside the base image of stati-web-server. So for now I just did a minor update

File: services/dy-static-file-server/src/dy_static_file_server/inputs_to_outputs.py [hunk 2]
```diff
@@ -112,8 +72,72 @@ def remap_input_to_output(input_dir: Path, output_dir: Path) -> None:
...
+            self._keep_running = False
+            await self._monitor_task
+            self._monitor_task = None
+
+
+async def _run_background_sync(input_dir: Path, output_dir: Path) -> None:
+    monitor = PortsMonitor(input_dir, output_dir)
+    await monitor.start()
+    while True:
+        await asyncio.sleep(1)
```
Comment (bisgaard-itis) [inline services/dy-static-file-server/src/dy_static_file_server/inputs_to_outputs.py:137]: Don't you need to add calls to the `monitor` here?

**Review by bisgaard-itis — APPROVED** [review]
Thanks for the effort. Just a few questions from my side

File: services/dy-static-file-server/src/dy_static_file_server/inputs_to_outputs.py (hunk 2 above)
Comment (GitHK) [reply services/dy-static-file-server/src/dy_static_file_server/inputs_to_outputs.py:137]: No, it's supposed to wait her forever.

**Pull request merged**